### PR TITLE
Support new account fields for protocol-19.

### PR DESCRIPTION
**Pull request by erika-sdf**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://developers.stellar.org/api/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [x] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [x] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Add `LedgerSequence` and `LedgerTime` as account fields.

### Why

To allow for CAP-21 changes to transaction validity

### CODE REVIEW

File: services/horizon/internal/db2/history/accounts_test.go [hunk 1]
```diff
@@ -87,6 +94,7 @@ func TestInsertAccount(t *testing.T) {
 	assert.Equal(t, "GAOQJGUAB7NI7K7I62ORBXMN3J4SSWQUQ7FOEPSDJ322W2HMCNWPHXFB", accounts[0].AccountID)
 	assert.Equal(t, int64(20000), accounts[0].Balance)
 	assert.Equal(t, int64(223456789), accounts[0].SequenceNumber)
+	assert.Equal(t, uint32(2345), accounts[0].SequenceLedger)
```
Comment (sreuland) [inline services/horizon/internal/db2/history/accounts_test.go:97]: super minor, is it worthwhile asserting on new SequenceTime value also?

File: protocols/horizon/main.go [hunk 2]
```diff
@@ -43,6 +43,8 @@ type Account struct {
 	ID                   string            `json:"id"`
 	AccountID            string            `json:"account_id"`
 	Sequence             string            `json:"sequence"`
+	SequenceLedger       uint32            `json:"sequence_ledger,omitempty"`
+	SequenceTime         *time.Time        `json:"sequence_time,omitempty"`
```
Comment (sreuland) [inline protocols/horizon/main.go:47]: @tamirms @Shaptic , there was some comment dialog that had taken place on [p19 doc](https://docs.google.com/document/d/1qZ6o14JBOZpyDG6rNEd8Bj-sCzpeZgyx7Vtl1SMKI28/edit#) about using epoch integrals(expressed as string on json resources) for timestamps such as `TimeBounds.MinTime`, wondering if Account.SequenceTime would follow the same type convention here? i.e. 

```
SequenceTime                    string `json:"sequence_time, omitempty"`
```

File: services/horizon/internal/ingest/processors/accounts_processor.go [hunk 3]
```diff
@@ -111,12 +111,29 @@ func (p *AccountsProcessor) ledgerEntryToRow(entry xdr.LedgerEntry) history.Acco
...
+	v1, found := account.Ext.GetV1()
+	if found {
+		v2, foundV2 := v1.Ext.GetV2()
+		if foundV2 {
+			v, foundV3 := v2.Ext.GetV3()
+			if foundV3 {
+				v3 = v
+			}
+		}
+	}
```
[the remark this answers is not in the document]
Comment (erika-sdf) [reply services/horizon/internal/ingest/processors/accounts_processor.go]: Definitely an improvement, have added this change.

File: services/horizon/internal/db2/history/accounts_test.go (hunk 1 above)
Comment (erika-sdf) [reply services/horizon/internal/db2/history/accounts_test.go:97]: Sure, have added a check for it.

File: protocols/horizon/main.go (hunk 2 above)
Comment (Shaptic) [reply protocols/horizon/main.go:47]: Yeah that's a great point. However, this string isn't user-controlled and I think the main justification for the timestamp component was the formatting snafus we saw with big times. I popped a quick thread into Slack to discuss

**Review by Shaptic — COMMENTED** [review]
Looks good so far! One thing we should definitely keep in mind and have tests around is the **absence** of V3 extension fields on an account, since these will only be filled, parsed, rendered, etc. on _new_ actions by an account.

File: services/horizon/internal/db2/schema/migrations/55_add_account_sequence_ledger.sql [hunk 4]
```diff
@@ -0,0 +1,7 @@
+-- +migrate Up
+ALTER TABLE accounts ADD sequence_ledger integer;
+ALTER TABLE accounts ADD sequence_time timestamp without time zone;
```
[the remark this answers is not in the document]
Comment (erika-sdf) [reply services/horizon/internal/db2/schema/migrations/55_add_account_sequence_ledger.sql]: Sure, have set them to explicitly default to null.

File: services/horizon/internal/ingest/processors/accounts_processor_test.go [hunk 5]
```diff
@@ -103,6 +105,23 @@ func (s *AccountsProcessorTestSuiteLedger) TestNewAccount() {
...
+							V: 3,
+							V3: &xdr.AccountEntryExtensionV3{
+								SeqLedger: 2345,
+								SeqTime:   1647265533,
+							},
+						},
+					},
+				},
+			},
+		},
```
[the remark this answers is not in the document]
Comment (erika-sdf) [reply services/horizon/internal/ingest/processors/accounts_processor_test.go:188]: Done

**Review by leighmcculloch — APPROVED** [review]

File: protocols/horizon/main.go (hunk 2 above)
Comment (Shaptic) [reply protocols/horizon/main.go:47]: @erika-sdf LGTM besides this last tidbit: lets make this a stringified int64 unix timestamp

Comment (erika-sdf) [reply protocols/horizon/main.go]: Sure. I've changed `SequenceTime` to be a string of seconds in `protocol/main`, but have left it in the `time.Time` representation for the db.

**Review by sreuland — APPROVED** [review]
looks good, thanks for adjustments on sequence time.

**Review by Shaptic — APPROVED** [review]

**Pull request merged**